cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/sql/PostgreSqlStructuredTable.java

```diff
@@ -60,16 +65,20 @@
  */
 public class PostgreSqlStructuredTable implements StructuredTable {
   private static final Logger LOG = LoggerFactory.getLogger(PostgreSqlStructuredTable.class);
-  private static final int SCAN_FETCH_SIZE = 100;
+  private static final String SCAN_FETCH_SIZE = "data.storage.sql.scan.size.rows";
+  private static final Injector injector = Guice.createInjector(new ConfigModule());
```

Comment:
Usually you don't create injector in the business logic classes. CConfiguration should be passed as argument in constructors chain up to to one annotated with `@Inject` (e.g. `io.cdap.cdap.spi.data.sql.RetryingSqlTransactionRunner`)

Reply:
I agree. I thought of doing this initially but looked like too many changes across multiple files. Involves changing 7 constructors.

PFB hierarchy
1. PostgreSqlStructuredTable -> SqlStructuredTableContext -> SqlTransactionRunner -> RetryingSqlTransactionRunner(has cconf)
2. PostgreSqlStructuredTable -> SqlStructuredTableContext -> SqlTransactionRunner -> SqlStructuredTableRegistry -> PostgreSqlStructuredTableAdmin -> PostgreSqlStorageProvider(has cconf)

SqlTransactionRunner is being initialized by both SqlStructuredTableRegistry and RetryingSqlTransactionRunner.
Feel free to let me know if we need to do it or if theres a better way to approach this. Thanks

Reply:
Yes, it's quite a bit of changes, but it also provides transparency on the dependencies. The current approach is not correct as you don't know full proper injector configuration and mixing up beans from two injectors is not correct ( you start getting duplicates of beans that are expected to be singletons)

Reply:
The other approach is using factories & assisted injection, but it's an overkill to add it just for this change

Reply:
The cConf is available via the PostgreSqlStorageProvider
You should extract configs you need from there (e.g. fetch size) and pass it to the PostgreSqlStructuredTableAdmin

Reply:
Ideally postgresql related classes should be implemented as extension, which it won't have access to core CDAP objects (e.g. cConf) and it shouldn't use guice injection to get hold of CDAP internal objects.

Reply:
See usage for other configurations (e.g. `data.storage.sql.jdbc.connection.pool.size`) as an example.

Reply:
Changed as per comments.